plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/BridgeVifDriver.java

```diff
@@ -287,7 +287,7 @@ private void deleteExitingLinkLocalRouteTable(String linkLocalBr) {
             for (String line : lines) {
                 String[] tokens = line.split(" ");
                 if (!tokens[2].equalsIgnoreCase(linkLocalBr)) {
```

Comment:
Consider the following as a bit of extra safety in the event of malformed input beginning at line 289:

```
if (tokens != null && tokens.length < 2) {
   continue;
}

final String device = tokens[2];
if (!Strings.isNullOrEmpty(device) &&
    !device.equalsIgnoreCase(linkLocalBr)) {

```